test/provider_file_memory.cpp

```diff
@@ -0,0 +1,243 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "base.hpp"
+
+#include "cpp_helpers.hpp"
+
+#include <umf/memory_provider.h>
+#include <umf/providers/provider_file_memory.h>
+
+using umf_test::test;
+
+#define FILE_PATH ((char *)"tmp_file")
+#define INVALID_PTR ((void *)0x01)
+
+#define ASSERT_IS_ALIGNED(ptr, alignment)                                      \
```

Comment:
I can see this macro is copy-pasted to all these providers - perhaps place it in some test helper file

Reply:
Done